### ISSUE TITLE
Low Rank transformation gives no output

### ISSUE THREAD

**Issue by SourabhRanade** [open-mmlab/mmdetection#6967]
Hello, I am currently using the Mask RCNN model with the SWIN transformer backbone.

I am trying a low-rank transformation(for the purpose of reducing the size of the model) on the model and want to run inference on it. However, the model returns nothing but the original image back. If I am right the inference_detector returns [xmin, ymin, xmax, ymax, class confidence score] (thus a total of five values). However, I don't see a bounding box on the output image.
I am not sure what the problem might be !  

I have done the low-rank transformation using svd  in pytorch as follows :

```
def replace_layers(model):
    svd_classifier_layers=[]
    L=40
    count=1
    for module_name, module in model.named_children():
      # print(module_name)
      if module_name=='proj':
        continue
      if len(list(module.children())) > 0:
        replace_layers(module)
      if isinstance(module, torch.nn.Linear):
        W=module.weight.data
        # print(W)
        # U,S,V=torch.svd(W)
        U, S, Vh = torch.linalg.svd(W,full_matrices=False)
        V = Vh.transpose(-2, -1).conj()
        W1 = U[:,:L]
        W2 = torch.diag(S[:L]) @ V[:,:L].t()
        if module.bias is not None:
          # print("W1:",W1)
          layer_1 = nn.Linear(in_features=module.in_features,
                            out_features=L, bias=True)
          layer_1.weight.data = W2
          svd_classifier_layers.append(layer_1)

          layer_2 = nn.Linear(in_features=L,
                            out_features=module.out_features , bias=True)
          layer_2.weight.data = W1
          layer_2.bias.data = module.bias.data
          svd_classifier_layers.append(layer_2)
          # print("Layer2",layer_2.weight.data)
          # break
        else:
          layer_1 = nn.Linear(in_features=module.in_features,
                            out_features=L, bias=False)
          layer_1.weight.data = W2
          svd_classifier_layers.append(layer_1)

          layer_2 = nn.Linear(in_features=L,
                            out_features=module.out_features , bias=False)
          layer_2.weight.data = W1
          # layer_2.bias.data = module.bias.data
          svd_classifier_layers.append(layer_2)

        try:
          # print("we are now trying to replace",module,"with",svd_classifier_layers,"in a sequnetial way")
          # n =int(module_name)
          # model[n]=nn.Sequential(*svd_classifier_layers)
          setattr(model, module_name, nn.Sequential(*svd_classifier_layers))
        except:
          setattr(model, module_name, nn.Sequential(*svd_classifier_layers))
    # return model
```

I am using the following for inference  : 

```
img_3='/content/mmdetection/Datasets/cocodataset/val2017/val2017/000000000632.jpg'

  # or img = mmcv.imread(img), which will only load it once
model.to('cpu')
result = inference_detector(model, img_3)
print(result)
# visualize the results in a new window
model.show_result(img_3, result)
# or save the visualization results to image files
model.show_result(img_3, result, out_file='result_tm_8.jpg')
```
**and a snippet of the  results are :** 
```
([array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([[0.0000000e+00, 5.7013138e+01, 6.4000000e+02, 4.8300000e+02,
        2.8487656e-01],
       [0.0000000e+00, 2.3504463e+02, 4.6826035e+02, 4.8300000e+02,
        8.4198877e-02]], dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32), array([], shape=(0, 5), dtype=float32)], [[], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [array([[False, False, False, ..., False, False, False],
       [False, False, False, ..., False, False, False],
       [False, False, False, ..., False, False, False],  and so on 

```

I am using the `/mmdetection/configs/swin/mask_rcnn_swin-t-p4-w7_fpn_1x_coco.py'` which internally accesses the base model as   `/models/mask_rcnn_r50_fpn.py`  where I see : 

```
rcnn=dict(
            score_thr=0.05,
            nms=dict(type='nms', iou_threshold=0.5),
            max_per_img=100,
            mask_thr_binary=0.5)))

```

Can changing the confidence score help in this case? 
I tried changing the `score_the` in the above snippet but that did not change anything? Should I be changing the confidence at some other place as well?

Any help or guidance in this matter would be really helpful.

**Comment by SourabhRanade** [comment]
I figured this out and it was an implementation issue from my end.Thank you.
